modules/stochastic_tools/include/surrogates/GaussianProcess.h

```diff
@@ -0,0 +1,67 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "SurrogateModel.h"
+#include "GaussianProcessTrainer.h"
+#include "Standardizer.h"
+#include <Eigen/Dense>
+#include "CovarianceInterface.h"
+
+class GaussianProcess : public SurrogateModel, public CovarianceInterface
+{
+public:
+  static InputParameters validParams();
+  GaussianProcess(const InputParameters & parameters);
+  virtual Real evaluate(const std::vector<Real> & x) const override;
+  virtual Real evaluate(const std::vector<Real> & x, Real & std) const;
+
+  virtual void setupCovariance(UserObjectName _covar_name);
```

Comment:
It isn't obvious how and when this is called. Could you add a comment on why and how it is used in the action?